### PR TITLE
feat: adjust the mat-button mat-card, mat-checkbox, mat-chips and mat-dialog components after upgrade to angular 15

### PR DESCRIPTION
**Pull request by xflord**

### CODE REVIEW

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
The ellipsis is not working in mat-chip:
image

File: libs/perun/components/src/lib/refresh-button/refresh-button.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <button
   (click)="onClickbutton($event)"
   [disabled]="disabled"
-  class="mr-2"
-  mat-stroked-button
+  class="mr-2 mb-1"
+  mat-icon-button
```
[the remark this answers is not in the document]
Comment (xflord) [reply libs/perun/components/src/lib/refresh-button/refresh-button.component.html:5]: The aligning of icons in buttons without text (unless its an icon button) is all messed up now. I spent a whole day trying to fix it and didn't manage to find a sensible solution. Will try a bit more though

File: apps/admin-gui/src/app/shared/components/dialogs/add-facility-owner-dialog/add-facility-owner-dialog.component.html [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-      (filter)="applyFilter($event)"
-      [placeholder]="'DIALOGS.ADD_OWNERS.FILTER'"
-      class="font-size-1rem"></perun-web-apps-immediate-filter>
-
+    <div mat-dialog-content>
+      <perun-web-apps-immediate-filter
+        (filter)="applyFilter($event)"
+        [placeholder]="'DIALOGS.ADD_OWNERS.FILTER'"
+        class="font-size-1rem"></perun-web-apps-immediate-filter>
+    </div>
```
[the remark this answers is not in the document]
Comment (xflord) [reply apps/admin-gui/src/app/shared/components/dialogs/add-facility-owner-dialog/add-facility-owner-dialog.component.html:10]: There are elements in which the filter stays on top, eg. the dialog for adding a new attribute (which is more user-friendly in my opinion). If we want to unify this then a new task should be made, however, I heavily lean towards this variant.

**Comment by xflord** [comment]
Fixed the buttons. Pretty sure the correct version of chips is on my computer at the office; I will stop by in the afternoon and fix it then.

**Comment by xflord** [comment]
Didn't manage to make the chips component (attribute value list) flexible as before even after racking my brains for 2 days already, I've fixed the ellipsis and made the width static for now (so we can potentially merge this PR and fix it later)

File: libs/perun/components/src/lib/attributes-list/attribute-value/attribute-value-list/attribute-value-list.component.html [hunk 3]
```diff
@@ -1,49 +1,53 @@
...
+  </div>
+  <div class="w-100" *ngIf="values.length > defaultItemsShown">
+    <span [matTooltip]="showMore ? 'Show less' : 'Show more'" [matTooltipPosition]="'above'">
+      <button class="ms-auto" mat-icon-button (click)="onShowChange()">
+        <mat-icon *ngIf="!showMore">more_horiz</mat-icon>
+        <mat-icon *ngIf="showMore">keyboard_arrow_up</mat-icon>
+      </button>
+    </span>
+  </div>
+  <mat-form-field *ngIf="!readonly" subscriptSizing="dynamic" class="w-100">
```
Comment (HejdaJakub) [inline libs/perun/components/src/lib/attributes-list/attribute-value/attribute-value-list/attribute-value-list.component.html:42]: On this page is an error when the attribute is read-only due to required `matChipInputFor` for `mat-chip-grid`.
image
You can just replace `*ngIf` by `[hidden]` and it should work correctly.
```suggestion
  <mat-form-field [hidden]="readonly" subscriptSizing="dynamic" class="w-100">
```

File: libs/perun/components/src/lib/back-button/back-button.component.html [hunk 4]
```diff
@@ -8,5 +8,5 @@
   class="mb-2 mt-2"
   (click)="goBack()"
   matTooltip="{{'Back' | translate}}">
-  <mat-icon class="rotate">arrow_right_alt</mat-icon>
+  <mat-icon class=" ml-auto rotate">arrow_right_alt</mat-icon>
```
Comment (HejdaJakub) [inline libs/perun/components/src/lib/back-button/back-button.component.html:11]: Again now it looks the same without this class I think.
```suggestion
  <mat-icon class="rotate">arrow_right_alt</mat-icon>
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]

**Comment by xflord** [comment]
Fixed what you mentioned along with checkbox labels that were affected by a new global CSS style

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**